marketdata/rest.go

```diff
@@ -97,13 +97,17 @@ func defaultDo(c *Client, req *http.Request) (*http.Response, error) {
 
 	var resp *http.Response
 	var err error
+
+RetryLoop:
 	for i := 0; ; i++ {
 		resp, err = c.httpClient.Do(req)
 		if err != nil {
 			return nil, err
 		}
-		if resp.StatusCode != http.StatusTooManyRequests {
-			break
+		switch resp.StatusCode {
+		case http.StatusTooManyRequests, http.StatusInternalServerError:
+		default:
+			break RetryLoop
```

Comment:
doesn't this mean it will now always retry up to the RetryLimit even on successes?

Reply:
No. On success the status code is not 429 or 500, so we will break out of the loop on this line.

Reply:
aah you're right. I misread and thought it'd continue here for some reason.